evm_loader/program/src/entrypoint.rs

```diff
@@ -275,6 +275,74 @@ fn process_instruction<'a>(
+            let account_info_iter = &mut accounts.iter();
+            let holder_info = next_account_info(account_info_iter)?;
+            let storage_info = next_account_info(account_info_iter)?;
+            let _program_info = next_account_info(account_info_iter)?;
+            let _program_code = next_account_info(account_info_iter)?;
+            let _caller_info = next_account_info(account_info_iter)?;
+
+            let  accounts = &accounts[1..];
+
+            let (unsigned_msg, signature) = {
+                let data = holder_info.data.borrow();
+                let account_info_data = AccountData::unpack(&data)?;
+                match account_info_data {
+                    AccountData::Empty => (),
+                    _ => return Err(ProgramError::InvalidAccountData),
+                };
+
+                let (acc_header, rest) = data.split_at(account_info_data.size());
+                let (signature, rest) = rest.split_at(65);
+                let (trx_len, rest) = rest.split_at(8);
+                let trx_len = trx_len.try_into().ok().map(u64::from_le_bytes).unwrap();
+                let (trx, _rest) = rest.split_at(trx_len as usize);
+                (trx.to_vec(), signature.to_vec())
+            };
+            let (nonce, to, data) = get_data(&unsigned_msg);
+            if let Err(e) = verify_tx_signature(&signature, &unsigned_msg) {
+                debug_print!("{}", e);
+                return Err(ProgramError::InvalidInstructionData);
+            }
+            let trx: UnsignedTransaction = rlp::decode(&unsigned_msg).map_err(|_| ProgramError::InvalidInstructionData)?;
+
+
+            let mut account_storage = ProgramAccountStorage::new(program_id, &accounts[1..], accounts.last().unwrap())?;
+
+            let caller = account_storage.get_caller_account().ok_or(ProgramError::InvalidArgument)?;
+            if caller.get_nonce() != nonce {
+                debug_print!("Invalid nonce: actual {}, expect {}", nonce, caller.get_nonce());
+                return Err(ProgramError::InvalidInstructionData);
+            }
+            let caller_ether = caller.get_ether();
+            debug_print!("   caller: {}", &caller_ether.to_string());
+
+            let mut storage = StorageAccount::new(storage_info, accounts, caller_ether, trx.nonce)?;
+
+            let backend = SolanaBackend::new(&account_storage, Some(accounts));
+            debug_print!("  backend initialized");
+
+            if trx.chain_id != backend.chain_id() {
+                debug_print!("Invalid chain id: actual {}, expect {}", trx.chain_id, backend.chain_id());
+                return Err(ProgramError::InvalidInstructionData);
+            }
+
+            let executor_state = ExecutorState::new(ExecutorSubstate::new(), backend);
+            let mut executor = Machine::new(executor_state);
+
+            debug_print!("Executor initialized");
+
+            executor.create_begin(caller_ether, data, u64::max_value());
```

Comment:
Before calling executor you should check that this is a deploy transaction and not a call transaction.